### PR TITLE
Provide Arrow Schema Hint to Parquet Reader

### LINKED ISSUE #5657: Provide Arrow Schema Hint to Parquet Reader
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

The parquet reader automatically uses an embedded arrow schema to hint type inference for decode. In particular if the hinted type is compatible with the underlying parquet type, it performs a cast.

**Describe the solution you'd like**

In situations where the writer was not an arrow writer this schema is not available, and therefore the arrow types are inferred from the parquet schema. This is not always desirable:

* https://github.com/apache/arrow-rs/issues/5655
* https://github.com/apache/arrow-rs/pull/5605#issuecomment-2060960080

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Lordworms**
# Which issue does this PR close?

Closes #5657 

# Rationale for this change
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by Lordworms** [comment]
seems like the integration process has some problem

File: arrow-schema/src/datatype.rs [hunk 1]
```diff
@@ -688,6 +688,32 @@ pub const DECIMAL256_MAX_SCALE: i8 = 76;
 /// values
 pub const DECIMAL_DEFAULT_SCALE: i8 = 10;
 
+// used for <https://github.com/apache/arrow-rs/pull/5605> we should decide what two different
+// Datatype could be reinterpret
+pub fn can_reinterpret(from_type: &DataType, to_type: &DataType) -> bool {
```
Comment (tustvold) [inline arrow-schema/src/datatype.rs:693]: This appears to be duplicating the existing logic for schema jnference using the arrow schema as a hint, I'm not sure why we need this?

**Comment by alamb** [comment]
FYI there is another alternative idea here: https://github.com/apache/arrow-rs/pull/5939 which looks promising

**Comment by tustvold** [comment]
Closed by https://github.com/apache/arrow-rs/pull/5939

**Pull request closed without merging**